lib/mintacoin_web/plugs/verify_api_token.ex

```diff
@@ -20,28 +21,32 @@ defmodule MintacoinWeb.Plugs.VerifyApiToken do
 
   @impl true
   def call(conn, _default) do
-    basic_token = get_token(conn)
-    valid_token = Application.get_env(:mintacoin, :api_token)
-
-    resolve_authorization(basic_token, valid_token, conn)
+    conn
+    |> get_token
+    |> Customers.verify_customer()
+    |> resolve_authorization(conn)
   end
 
-  @spec resolve_authorization(token :: token(), token :: token(), conn :: conn()) :: conn()
-  defp resolve_authorization(token, token, conn), do: conn
+  @spec resolve_authorization(params :: tuple(), conn :: conn()) :: conn()
+  defp resolve_authorization({:ok, token}, conn) do
+    %{customer_id: customer_id} = token
```

Comment:
A suggestion of style: 

```suggestion
  defp resolve_authorization({:ok, %{customer_id: customer_id}}, conn) do
```